plugin/src/patcher.ml4

```diff
@@ -103,42 +107,68 @@ let invert_patch n env patch sigma =
     failwith "Could not find a well-typed inverted term"
 
 (* Common patch command functionality *)
-let patch env n try_invert a search sigma =
+    
+let patch env n try_invert a search sigma define =
   let reduce = try_reduce reduce_remove_identities in
   let sigma, patch_to_red = search env a sigma in
   let sigma, patch = reduce env sigma patch_to_red in
   let prefix = Id.to_string n in
-  ignore (define_term n sigma patch false);
   (if !opt_printpatches then
     print_patch env sigma prefix patch
   else
     Printf.printf "Defined %s\n" prefix);
-  if try_invert then
+  (if try_invert then
     try
       let inv_n_string = String.concat "_" [prefix; "inv"] in
       let inv_n = Id.of_string inv_n_string in
       invert_patch inv_n env patch sigma
     with _ ->
       ()
   else
-    ()
+    ());
+  define sigma patch
```

Comment:
In which case, the `define` call here should go before the call to `invert_patch`